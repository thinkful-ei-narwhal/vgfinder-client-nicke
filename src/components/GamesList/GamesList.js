import "./GamesList.css";
import React, { Component } from "react";
import GameItem from "../GameItem/GameItem";

export default class GamesList extends Component {
  static defaultProps = {
    gamesList: [],
  };

  render() {
    const gamesList = this.props.gamesList;
    return gamesList.map((game, index) => <GameItem key={index} game={game} />);
  }
}
