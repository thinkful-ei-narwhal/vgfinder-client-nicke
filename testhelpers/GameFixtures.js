const GameFixtures = {
  stubGameData: {
    id: "1",
    title: "Hollow Knight",
    description:
      "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
    genre: "Adventure",
    rating: "4",
    developer: "Team Cherry",
    trailer_url:
      "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
    image_url_box_art:
      "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
    image_url_two:
      "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
    image_url_three:
      "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
  },
  stubGameListData: [
    {
      id: "1",
      title: "Hollow Knight",
      description:
        "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
      genre: "Adventure",
      rating: "4",
      developer: "Team Cherry",
      trailer_url:
        "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      image_url_box_art:
        "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      image_url_two:
        "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
      image_url_three:
        "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
    },
    {
      id: "2",
      title: "Hollow Knight",
      description:
        "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
      genre: "Adventure",
      rating: "4",
      developer: "Team Cherry",
      trailer_url:
        "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      image_url_box_art:
        "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      image_url_two:
        "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
      image_url_three:
        "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
    },
  ],
};

export default GameFixtures;
