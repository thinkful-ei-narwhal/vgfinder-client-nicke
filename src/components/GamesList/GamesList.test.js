import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import GamesList from "./GamesList";
import GameFixtures from "./../../../testhelpers/GameFixtures";

describe(`GamesList component`, () => {
  it("renders a GamesList without props", () => {
    const wrapper = shallow(<GamesList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders a GamesList with supplied games", () => {
    const wrapper = shallow(
      <GamesList gamesList={GameFixtures.stubGameListData} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
