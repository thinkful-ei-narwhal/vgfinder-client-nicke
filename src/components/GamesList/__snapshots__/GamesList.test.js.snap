// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesList component renders a GamesList with supplied games 1`] = `
Array [
  <GameItem
    game={
      Object {
        "description": "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
        "developer": "Team Cherry",
        "genre": "Adventure",
        "id": "1",
        "image_url_box_art": "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
        "image_url_three": "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
        "image_url_two": "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
        "rating": "4",
        "release_date": "2018-07-22",
        "title": "Hollow Knight",
        "trailer_url": "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      }
    }
    key="0"
  />,
  <GameItem
    game={
      Object {
        "description": "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
        "developer": "Team Cherry",
        "genre": "Adventure",
        "id": "2",
        "image_url_box_art": "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
        "image_url_three": "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
        "image_url_two": "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
        "rating": "4",
        "release_date": "2018-07-22",
        "title": "Hollow Knight",
        "trailer_url": "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      }
    }
    key="1"
  />,
]
`;

exports[`GamesList component renders a GamesList without props 1`] = `null`;
